frictionless/errors/row.py

```diff
@@ -66,11 +66,72 @@ class PrimaryKeyError(RowError):
     template = 'Row at position "{rowNumber}" violates the primary key: {note}'
 
 
-class ForeignKeyError(RowError):
+@attrs.define(kw_only=True)
+class ForeignKeyError(TableError):
     type = "foreign-key"
     title = "ForeignKey Error"
     description = "Values in the foreign key fields should exist in the reference table"
     template = 'Row at position "{rowNumber}" violates the foreign key: {note}'
+    tags = ["#table", "#row"]
+
+    cells: List[str]
+    """NOTE: add docs"""
+
+    row_number: int
+    """NOTE: add docs"""
+
+    target_keys: List[str]
+    """NOTE: add docs"""
+
+    source_keys: List[int]
```

Comment:
List[str]?

Reply:
I turned into an Iterable, I have seen applications of the library where the index values are strings and so far it has been working properly.